plugins/cloud-aws/src/main/java/org/elasticsearch/cloud/aws/blobstore/S3BlobStore.java

```diff
@@ -71,13 +71,28 @@ public S3BlobStore(Settings settings, AmazonS3 client, String bucket, @Nullable
         }
 
         this.numberOfRetries = maxRetries;
-        if (!client.doesBucketExist(bucket)) {
-            if (region != null) {
-                client.createBucket(bucket, region);
-            } else {
-                client.createBucket(bucket);
+
+        int retry = 0;
+        while (retry <= this.numberOfRetries) {
+            try {
+                if (!client.doesBucketExist(bucket)) {
+                    if (region != null) {
+                        client.createBucket(bucket, region);
+                    } else {
+                        client.createBucket(bucket);
+                    }
+                }
+                break;
+            } catch (AmazonClientException e) {
+                if (shouldRetry(e) && retry < this.numberOfRetries) {
```

Comment:
Thanks @tlrx.
> We're doing one extra execution here... I think we should have while (retry < maxRetries) and if (shouldRetry(e) && retry < (maxRetries - 1)).

I think we are doing one more try for all our s3 request [here], should we change them back?(https://github.com/elastic/elasticsearch/blob/4c981ff4bfc250080d521af105b5e8589c9fc517/plugins/cloud-aws/src/main/java/org/elasticsearch/cloud/aws/blobstore/DefaultS3OutputStream.java#L106)